lisp/ess-r-package.el

```diff
@@ -321,34 +348,44 @@ the other hand, `.metadata' is always ignored by the R package
 builder, which makes it a safe directory to store the revdep
 checking results."
   (interactive "P")
-  (ess-r-package-send-process ess-r-devtools-revdep-check-cmd
-                              "Checking reverse dependencies of %s"
-                              alt))
+  (ess-r-package-eval-linewise
+   ess-r-devtools-revdep-check-cmd "Checking reverse dependencies of %s"))
 
-(defun ess-r-devtools-document-package (&optional alt)
-  "Interface for `devtools::document()'."
+(defun ess-r-devtools-document-package (&optional arg)
+  "Interface for `devtools::document()'.
+With prefix ARG ask for extra arguments."
   (interactive "P")
-  (ess-r-package-send-process "devtools::document(%s)\n"
-                              "Documenting %s"
-                              alt))
-
-(defun ess-r-devtools-install-package (&optional alt)
-  "Interface to `devtools::install()'."
+  (ess-r-package-eval-linewise
+   "devtools::document(%s)\n" "Documenting %s"
+   '("" (read-string "Document args: "))))
+
+(defun ess-r-devtools-install-package (&optional arg)
+  "Interface to `devtools::install()'.
+On prefix ARG (C-u), build the package first outside of current
+tree (local = FALSE). With double prefix ARG (C-u C-u) install
+quickly (quick = TRUE, upgrade_dependencies = FALSE)."
   (interactive "P")
-  (ess-r-package-send-process "devtools::install(%s)\n"
-                              "Installing %s"
-                              alt))
-
-(defun ess-r-devtools-install-github (&optional alt repo)
+  (ess-r-package-eval-linewise
+   "devtools::install(%s)\n" "Installing %s" arg
+   '("keep_source = TRUE"
+     (read-string "Install args: " "keep_source = TRUE, local = FALSE")
```

Comment:
When one want to re-build the package from scratch (don't re-use dlls). There is no `cleanup` argument. 

Keep source works in this case if the package is in `ess-r-package-library-paths`.

Reply:
FYI I'm using `load_all(recompile = TRUE)` prior to installing to achieve this.

Reply:
Right, recompile=TRUE is on C-u and local=FALSE is the closest alternative. 

I am staying away from load_all because it didn't work well in the past and it messed with help files a big deal. But maybe I should give it one more chance.